### DISCUSSION TITLE
Vercel with GitLab

### DISCUSSION THREAD

**Discussion by AmanRaj1608** [vercel/vercel#5578] [Help] [answered]
I have a GitLab CI pipeline set up which test my linting.
I want my vercel to deeply only after that pipeline.

How can handle vercel deployment gitlab pipeline.

**Comment by maxstoyanov** [comment, marked as answer]
There are two ways:

1. **Run the linting during the Vercel build process**:
You could edit your build command to include the linting process `npm run linting && npm run build`. If you have an integration between the linting tool and GitLab for displaying the results you will lose that since Vercel only displays the build logs.

2. **Deploy from GitLab CI with the Vercel CLI**
In this case you'd run the `vercel deploy` command as second step in GitLab CI. Look at the manuals for Vercel CLI and GitLab CI for their respective best practices.

**Reply by AmanRaj1608** [reply]
Can you please elaborate?
Like adding vercel-cli to my GitLab `.yaml` file?

**Reply by maxstoyanov** [reply]
I updated my answer to be more specific. Does that help you?
